security/v1/types.go

```diff
@@ -93,6 +93,14 @@ type SecurityContextConstraints struct {
 	AllowHostPID bool `json:"allowHostPID" protobuf:"varint,11,opt,name=allowHostPID"`
 	// AllowHostIPC determines if the policy allows host ipc in the containers.
 	AllowHostIPC bool `json:"allowHostIPC" protobuf:"varint,12,opt,name=allowHostIPC"`
+	// AllowHostUsers determines if the policy allows host users in the containers.
+	// +openshift:enable:FeatureGate=UserNamespacesSupport
+	// +kubebuilder:default="true"
+	// +kubebuilder:validation:Enum="true";"false";""
```

Comment:
I think you missed my point about not using bools 😂 

We avoid bools because they are not expressive and do not allow future expansion. What you've done here is gamed the system to force a bool into an enum.

The idea would be that you come up with a descriptive field name, and some enum values that make sense, such that if we needed additional states in the future, we could add those.

It would be more like
```
hostUsers: Allowed | Disallowed | AllowedButOnlyIfXYZ | DisallowedExceptForWhenFoo
```

Reply:
I thought the UX would be better if it could be expressed the same as the other AllowHost* fields in YAML. We still get the same level of expressiveness as we can in the future add AllowedButOnlyIfXYZ, but we maintain compatibility with existing fields. Are you against this approach??

Reply:
We try not to repeat sins of the past when we know better, and even with the enum, using true and false would break compatibility with existing types as `true != "true"` when you're parsing JSON

Reply:
Okay I've switched to Allowed/Disallowed. Personally, I don't see a difference, but I'll defer to your wisdom here :)